File: Extra - Estudos/2654 - Godofor/main.c
#include <stdio.h>
#include <string.h>
#include <stdlib.h>

typedef struct{
    char nome[1000];
    int poder;
    int matou;
    int morreu;
} Seres;


void sortGanhador(Seres s[], int c){

    for(int i = 0 ; i < c - 1 ; i++){
        int maior = i;
        for(int j = i + 1 ; j < c ; j++){
            if(s[maior].poder < s[j].poder){
                maior = j;
            }  else if(s[maior].poder == s[j].poder){
                if(s[maior].matou < s[j].matou){
                    maior = j;
                } else if(s[maior].matou == s[j].matou){
                    if(s[maior].morreu > s[j].morreu){
                        maior = j;
                    } else if(s[maior].morreu == s[j].morreu){
                        if(strcmp(s[maior].nome, s[j].nome) > 0){
                            maior = j;
                        }
                    }
                }    
            }
        }
        Seres tmp = s[i];
        s[i] = s[maior];
        s[maior] = tmp;
    }   
    printf("%s\n", s[0].nome);
}

int main (){
    int concorrentes;

    scanf("%d", &concorrentes);
    getchar();

    Seres s[concorrentes];

    for(int i = 0 ; i < concorrentes ; i++){
        scanf("%s %d %d %d", s[i].nome, &s[i].poder, &s[i].matou, &s[i].morreu);
    }

    sortGanhador(s, concorrentes);

    return 0;
}

File: Extra - Estudos/2654 - Godofor/Main.java
import java.util.*;

class Seres{
    String nome;
    int poder;
    int matou;
    int mortes;

    public Seres(String nome, int poder, int matou, int mortes){
        this.nome = nome;
        this.poder = poder;
        this.matou = matou;
        this.mortes = mortes;
    }
}


public class Main {
    public static void sortWinner(Seres[] seres, int n){
        for(int i = 0; i < n - 1 ; i++){
            int maior = i;
            for(int j = i + 1; j < n ; j++){
                if(seres[maior].poder < seres[j].poder){
                    maior = j;
                } else if (seres[maior].poder == seres[j].poder){
                    if(seres[maior].matou < seres[j].matou){
                        maior = j;
                    } else if (seres[maior].matou == seres[j].matou){
                        if(seres[maior].mortes > seres[j].mortes){
                            maior = j;
                        } else if (seres[maior].mortes == seres[j].mortes){
                            if(seres[j].nome.compareTo(seres[maior].nome) < 0){
                                maior = j;
                            }
                        }
                    }
                }
            }
            Seres tmp = seres[i];
            seres[i] = seres[maior];
            seres[maior] = tmp;

        }
    }

    public static void main(String[] args){
        Scanner sc = new Scanner(System.in);

        int numPlayers = sc.nextInt();
        sc.nextLine();

        Seres[] seres = new Seres[numPlayers];

        for(int i = 0 ; i < numPlayers ; i++){
            String nome = sc.next();
            int poder = sc.nextInt();
            int matou = sc.nextInt();
            int mortes = sc.nextInt();
            seres[i] = new Seres(nome, poder, matou, mortes);
        }

        sortWinner(seres, numPlayers);
        System.out.println(seres[0].nome);

        sc.close();
    }
}
